src/piecelink/download.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Download and validate utilities for PieceLink pieces
+ *
+ * This module provides functions to download data from a Response object,
+ * calculate PieceLink during streaming, and validate it matches the expected value.
+ */
+
+import type { PieceLink } from './index.js'
+import { asPieceLink, createPieceLinkStream } from './index.js'
+
+/**
+ * Download data from a Response object, validate its PieceLink, and return as Uint8Array
+ *
+ * This function:
+ * 1. Streams data from the Response body
+ * 2. Calculates PieceLink during streaming
+ * 3. Collects all chunks into a Uint8Array
+ * 4. Validates the calculated PieceLink matches the expected value
+ *
+ * @param response - The Response object from a fetch() call
+ * @param expectedPieceLink - The expected PieceLink to validate against
+ * @returns The downloaded data as a Uint8Array
+ * @throws Error if PieceLink validation fails or download errors occur
+ *
+ * @example
+ * ```typescript
+ * const response = await fetch(url)
+ * const data = await downloadAndValidatePieceLink(response, 'baga6ea4seaq...')
+ * ```
+ */
+export async function downloadAndValidatePieceLink (
```

Comment:
Nitpick, feel free to ignore.

When I read this name the first time, I parsed it as "(download and validate) (piece link)" and I was wondering how you can download a (piece) link.

I would find this name easier to understand if there were a noun after "download", e.g. "downloadBodyAndValidatePieceLink" or "downloadPieceAndValidateIntegrity".

Reply:
:ack: how about just `downloadAndValidate`, keep it short and sweet